ddtrace/internal/ci_visibility/recorder.py

```diff
@@ -977,9 +977,6 @@ def _register_module_handlers():
 def _on_discover_suite(discover_args: CISuite.DiscoverArgs):
     log.debug("Handling discovery for suite args %s", discover_args)
     module = CIVisibility.get_module_by_id(discover_args.suite_id.parent_id)
-    if discover_args.suite_id in module.children:
```

Comment:
@romainkomorndatadog Just a question: this affects both the current and the new version, right?

Reply:
It does not, because the current version does not use this API.

I'm removing this because there's already another check in [CIVisibilityParentItem.add_child()](https://github.com/DataDog/dd-trace-py/blob/main/ddtrace/internal/ci_visibility/api/ci_base.py#L492-L498) that does a similar thing (but only if `self._session_settings.reject_duplicates` ).

This change does mean we unnecessarily pay the cost penalty of `CIVisibilitySuite.__init__()`, but that's an optimization I'm willing to leave to later.